### ISSUE TITLE
[BUG] pydantic custom root models error when generating a spec

### ISSUE THREAD

**Issue by kageurufu** [0b01001001/spectree#254]
**Describe the bug**
 
When using a pydantic custom `__root__`, the base model has no properties, and errors when generating the spec.

**To Reproduce**

```python

import typing

import flask
import pydantic
import spectree

app = flask.Flask(__name__)
spec = spectree.SpecTree("flask", app=app)


class A(pydantic.BaseModel):
    a: typing.Literal["a"]


class B(pydantic.BaseModel):
    b: typing.Literal["b"]


class Req(pydantic.BaseModel):
    __root__: typing.Union[A, B]


@app.route("/test")
@spec.validate(query=Req)
def test():
    pass


with app.app_context():
    print(spec.spec)
```

**Expected behavior**

The openapi spec should generate, showing the discriminated union types

**Error Message**

```
Traceback (most recent call last):
  File "/home/frank/comanage-src/comanage/repro.py", line 32, in <module>
    print(spec.spec)
  File "/home/frank/.cache/pypoetry/virtualenvs/comanage-6JkL5Bll-py3.10/lib/python3.10/site-packages/spectree/spec.py", line 92, in spec
    self._spec = self._generate_spec()
  File "/home/frank/.cache/pypoetry/virtualenvs/comanage-6JkL5Bll-py3.10/lib/python3.10/site-packages/spectree/spec.py", line 280, in _generate_spec
    "parameters": parse_params(func, parameters[:], self.models),
  File "/home/frank/.cache/pypoetry/virtualenvs/comanage-6JkL5Bll-py3.10/lib/python3.10/site-packages/spectree/utils.py", line 109, in parse_params
    for name, schema in model["properties"].items():
KeyError: 'properties'
```

**Python Information (please complete the following information):**
 - Python Version 3.10
 - Library Version 0.10.5
 - Other dependencies: flask

**Comment by kemingy** [comment]
Thanks for your feedback.

It looks like the `__root__` is not handled properly. I'll create a PR to fix this.

**Comment by kemingy** [comment]
Hi @kageurufu, this has been fixed in the new release v0.10.6 or v1.0.0a1. You can try to upgrade. Feel free to reopen it if there is any problem.

**Comment by kageurufu** [comment]
wow, thank you so much! I was going to dig into this more today myself
